backend/src/main/java/com/capstone/moneytree/validator/UserValidator.java

```diff
@@ -0,0 +1,80 @@
+package com.capstone.moneytree.validator;
+
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.capstone.moneytree.dao.UserDao;
+import com.capstone.moneytree.exception.MissingMandatoryFieldException;
+import com.capstone.moneytree.handler.exception.UserAlreadyExistsException;
+import com.capstone.moneytree.model.node.User;
+
+@Component
+public class UserValidator implements Validator {
+   private static final Logger LOG = LoggerFactory.getLogger(UserValidator.class);
+
+   private final UserDao userDao;
+
+   @Autowired
+   public UserValidator(UserDao userDao) {
+      this.userDao = userDao;
+   }
+
+   @Override
+   public boolean supports(Class<?> aClass) {
```

Comment:
Just out of curiousity, when would this validation be useful?

Reply:
For now I think probably not much use for us, but it has to be overriden since its part of the Validator interface. I searched a bit and found this [doc link](https://docs.spring.io/spring-framework/docs/4.1.7.RELEASE/spring-framework-reference/html/validation.html) where it explains bean validation. There's a nice example where they check in the constructor if the validator class supports the object passed, which is good to return more meaningful error messages. The support can be implemented to either return true for the class itself, or tweaked to allow reuse with other classes and/or inheritance stuff. Good comment.

Reply:
https://github.com/DPigeon/Money-Tree/pull/163/commits/6cb6a7711029b596897c6465830a57f2f9931112